### PR TITLE
Entity Selector Option API

### PR DESCRIPTION
**Pull request by apple502j**
Adds an entity selector option API.

### Is this a port from (insert another API library)?
No. The underlying design might be similar (after all, this is just a wrapper method), but the code is written entirely by me. This also adds an easy way to make a non-repeatable selector.

### Is this tested?
Yes, actually this is tested using GameTest!

### CODE REVIEW

File: fabric-command-api-v2/src/main/java/net/fabricmc/fabric/api/command/v2/FabricEntitySelectorReader.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * or non-repeatable entity selector option.
+ */
+public interface FabricEntitySelectorReader {
+	/**
+	 * Sets a flag.
+	 * @param key the key of the flag
+	 * @param value the value of the flag
+	 */
+	default void setCustomFlag(Identifier key, boolean value) {
+		throw new UnsupportedOperationException("Not implemented");
```
Comment (Technici4n) [inline fabric-command-api-v2/src/main/java/net/fabricmc/fabric/api/command/v2/FabricEntitySelectorReader.java]:
```suggestion
		throw new UnsupportedOperationException("Implemented via mixin");
```

File: fabric-command-api-v2/src/main/java/net/fabricmc/fabric/api/command/v2/FabricEntitySelectorReader.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	/**
+	 * Gets the value of the flag.
+	 * @param key the key of the flag
+	 * @return the value, or {@code false} if the flag is not set
+	 */
+	default boolean getCustomFlag(Identifier key) {
+		throw new UnsupportedOperationException("Not implemented");
```
Comment (Technici4n) [inline fabric-command-api-v2/src/main/java/net/fabricmc/fabric/api/command/v2/FabricEntitySelectorReader.java]:
```suggestion
		throw new UnsupportedOperationException("Implemented via mixin");
```

File: fabric-command-api-v2/src/main/java/net/fabricmc/fabric/mixin/command/EntitySelectorOptionsAccessor.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.spongepowered.asm.mixin.gen.Invoker;
+
+import net.minecraft.command.EntitySelectorOptions;
+import net.minecraft.command.EntitySelectorReader;
+import net.minecraft.text.Text;
+
+@Mixin(EntitySelectorOptions.class)
+public interface EntitySelectorOptionsAccessor {
+	@Invoker
+	static void callPutOption(String id, EntitySelectorOptions.SelectorHandler handler, Predicate<EntitySelectorReader> condition, Text description) {
```
Comment (Technici4n) [inline fabric-command-api-v2/src/main/java/net/fabricmc/fabric/mixin/command/EntitySelectorOptionsAccessor.java:31]: Funny how the two classic types of options are called "call options" and "put options" in finance. :smile:

**Review by Technici4n — CHANGES_REQUESTED** [review]

**Review by Technici4n — APPROVED** [review]

**Comment by deirn** [comment]
I noticed that you can use Identifier as the key, but it needs to be quoted for it to work. Should we do that instead of replacing it with underscores? Might need to hack around the autocomplete suggestion though.

**Comment by apple502j** [comment]
@deirn I don't think that's very useful.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**